### PR TITLE
Fixes: #1200 'requirements.txt for testing dependency'

### PR DESCRIPTION
**Pull request by mayanksingh081**
PTAL @ezyang @fmassa

### CODE REVIEW

**Comment by ezyang** [comment]
`requirements.txt` doesn't imply anything for setup.py interaction, e.g. https://stackoverflow.com/questions/43658870/requirements-txt-vs-setup-py

But if you're afraid of people getting confused, you can always move the file to test.

This is kind of a much larger list than I expected though; also, I don't think we want the version pin on PyTorch.

**Comment by code-monk08** [comment]
@ezyang okay then should I remove the version for all the dependencies? also which dependencies should we keep?

**Comment by ezyang** [comment]
I'm not too familiar with torchvision's deps, @fmassa will have to answer that question :)

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+requests==2.9.1
+Pillow==6.1.0
+apex==0.9.10dev
+av==6.2.0
+lmdb==0.97
+mock==3.0.5
+pycocotools==2.0.0
+scipy==1.3.1
+torch==1.2.0
+chartio==6.0.1
```
Comment (fmassa) [inline requirements.txt]: Where is this used?

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,19 @@
+docutils==0.15.2
+matplotlib==1.5.1
+Jinja2==2.8
+pandas==0.24.2
+six==1.10.0
+sphinx_rtd_theme==0.4.3
```
Comment (fmassa) [inline requirements.txt]: This is only needed for building the docs

File: requirements.txt [hunk 3]
```diff
@@ -0,0 +1,19 @@
+docutils==0.15.2
+matplotlib==1.5.1
+Jinja2==2.8
```
Comment (fmassa) [inline requirements.txt]: This is only needed for building the binaries / nightlies

File: requirements.txt [hunk 4]
```diff
@@ -0,0 +1,19 @@
+docutils==0.15.2
```
Comment (fmassa) [inline requirements.txt]: I think this is only needed for building the docs

File: requirements.txt [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+matplotlib==1.5.1
+Jinja2==2.8
+pandas==0.24.2
+six==1.10.0
+sphinx_rtd_theme==0.4.3
+setuptools==20.7.0
+Sphinx==1.7.9
```
Comment (fmassa) [inline requirements.txt]: This is only needed for the docs

File: requirements.txt (hunk 1 above)
Comment (code-monk08) [reply requirements.txt]: - **torch** is listed as optional dependency in `hubconf.py`

- **scipy** is used in `setup.py`

- **pycocotools** is used in these three files `references/detection/coco_eval.py`, `torchvision/datasets/coco.py`, `references/detection/coco_utils.py`

- **chartio** : I used [pipreqs](https://github.com/bndr/pipreqs) to generate requirements.txt, I can't find where is chartio used.. I checked in sanity_checks.ipynb also.

File: requirements.txt (hunk 2 above)
Comment (code-monk08) [reply requirements.txt]: Removed.

File: requirements.txt (hunk 3 above)
Comment (code-monk08) [reply requirements.txt]: Removed.

File: requirements.txt (hunk 4 above)
Comment (code-monk08) [reply requirements.txt]: Removed.

File: requirements.txt (hunk 5 above)
Comment (code-monk08) [reply requirements.txt]: Removed.

**Comment by code-monk08** [comment]
Now I am only left with these modules.

- numpy
- requests
- Pillow
- apex
- av
- lmdb
- mock

should I finalize changes with these modules now? or should I add any from pycocotools, 
scipy, torch, chartio.

**Pull request closed without merging**